doc/migrations/v1.0.0-v2.0.0.md

```diff
@@ -0,0 +1,245 @@
+addresses of a range of peers in the ID space outside of the first KAD-bucket
+and not have a need to be constantly connected to them.
+
+To that end individual protocols can ensure that they have sufficient
+connections for their purposes and libp2p will not try to second-guess their
+rules to apply at a global level.
+
+**Before**
+
+```ts
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  // other config
+  connectionManager: {
+    // configure the autodialer
+    minConnections: 10
+  }
+})
+```
+
+**After**
+
+```ts
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  // other config
+})
+```
+
+## "Transient" connections have been renamed "limited"
+
+In libp2p terms, a `transient` connection is one that has limits applied to it
+for how long it's expected to remain open and/or how much data can be
+transferred over it.
+
+The most common place this type of connection will be encountered of this is one
+that is via a [v2 Circuit Relay](https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md).
+
+The actual limits are applied by the relay server and reported to both ends of
+the connection. They are optional so the connection may not have any limits and
+could be treated as a regular connection.
+
+The intention is for network nodes to run v2 Circuit Relay servers as a public
+good that allows nodes to create direct connections via, for example WebRTC.
+
+To prevent abuse, a relay will normally apply strict time/transfer limits, and
+only have a limited number of relay slots so as to not run as an open relay.
+
+`go-libp2p@0.34.0` included a breaking change of renaming `transient`
+connections as `limited`.
+
+`js-libp2p@2.0.0` adopts the same nomenclature so we can be sure we're talking
+about the same thing.
+
+**Before**
+
+```ts
+import { createLibp2p } from 'libp2p'
+import { multiaddr } from '@multiformats/multiaddr'
+
+const node = await createLibp2p({
+  // other config
+})
+
+// dial a peer
+const conn = await node.dial(multiaddr('/ip4/...'), {
+  signal: AbortSignal.timeout(1000)
+})
+
+// is the connection transient?
+console.info(conn.transient) // true | false
+
+// register a protocol handler for incoming protocol dials
+node.handle('/my-protocol/1.0.0', ({ stream, connection }) => {
+  // handle protocol
+}, {
+   runOnTransientConnection: true
+})
+
+// register a topology that is notified of peers connecting over transient
+// connections
+node.register('/my-protocol/1.0.0', {
+  onConnect: (peer, connection) => {
+    // handle connect
+  },
+  notifyOnTransient: true
+})
+```
+
+**After**
+
+```ts
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  // other config
+})
+
+// dial a peer
+const conn = await node.dial(multiaddr('/ip4/...'), {
+  signal: AbortSignal.timeout(1000)
+})
+
+// is the connection limited?
+console.info(conn.limits?.seconds) // number | undefined
+console.info(conn.limits?.bytes) // bigint | undefined
```

Comment:
will `conn.limits == null` if there are no limits? we should provide a snippet for devs wanting a similar `conn.transient` check

```suggestion
// is the connection limited?
console.info(conn.limits != null) // true = limited

// if it's limited, how is it limited?
console.info(conn.limits?.seconds) // number | undefined
console.info(conn.limits?.bytes) // bigint | undefined
```